vault/request_handling.go

```diff
@@ -434,10 +434,13 @@ func (c *Core) CheckToken(ctx context.Context, req *logical.Request, unauth bool
 		auth.PolicyResults.GrantingPolicies = append(auth.PolicyResults.GrantingPolicies, authResults.SentinelResults.GrantingPolicies...)
 	}
 
+	c.activityLogLock.RLock()
```

Comment:
I wonder if there is any subtle behaviour difference that the ENT side is implemented differently. Could you make the ENT side look like this and see if there is any behaviour change in tests or anything?

Reply:
I think the ENT side is good because I don't see a reason to slow down other threads just because we want to do some task inside the `HandleTokenUsage`.